api/src/main/java/edu/wpi/first/shuffleboard/api/widget/Widgets.java

```diff
@@ -58,80 +60,80 @@ public static void setDefault(Widgets instance) {
   }
 
   @Override
-  public void register(Class<? extends Widget> widgetClass) {
-    requireNonNull(widgetClass, "widgetClass");
-    if (isRegistered(widgetClass)) {
-      throw new IllegalArgumentException("Widget class " + widgetClass.getName() + " is already registered");
-    }
-    if (!widgetClass.isAnnotationPresent(Description.class)) {
-      throw new InvalidWidgetException(
-          "No description present on widget class " + widgetClass.getName());
+  public void register(ComponentType type) {
+    if (widgets.containsKey(type.getName())) {
+      throw new IllegalArgumentException("Component class " + type.getClass().getName() + " is already registered");
     }
+    widgets.put(type.getName(), type);
+    addItem(type);
+  }
+
+  /**
+   * Convenience overload for registering annotated widgets.
+   */
+  public <T extends Widget> void register(Class<T> widgetClass) {
```

Comment:
It would streamline this API at the cost of making the Plugins API *much* harder to work with. I'm not sure it's worth it (especially since you'd also have to do something about `registerDefaultWidget`, and that case seems much less obvious....)

Reply:
Not sure what the point of requiring AnnotatedWidget would be either, since it's not required to use the `@Description` annotation—it's just a helper.

Reply:
`AnnotatedWidget` requires a `@Description` annotation. Subclasses are out of spec if they don't have that annotation. `@ParamatrizedController` is optional. It's possible to implement `Widget` and use a `@Description` annotation, but if it's to be used they'll be reimplementing `AnnotatedWidget` anyway.

This method will also throw an exception if the class _isn't_ annotated with `@Description`, so it's better for compile-time safety to limit this to `Class<? extends AnnotatedWidget>`

Reply:
> It's possible to implement Widget and use a @Description annotation, but if it's to be used they'll be reimplementing AnnotatedWidget anyway.

there are reasons people would want to do that (multiple inheritance, for example)

if we don't need the actual type (and we don't) why require it?